src/github.com/getlantern/flashlight/logging/logging.go

```diff
@@ -410,3 +401,27 @@ func enableBorda(bordaReportInterval time.Duration, bordaSamplePercentage float6
 
 	ops.RegisterReporter(reporter)
 }
+
+func includeInSample(deviceID string, samplePercentage float64) bool {
+	if samplePercentage == 0 {
+		return false
+	}
+
+	// Sample a subset of device IDs.
+	// DeviceID is expected to be a Base64 encoded 48-bit (6 byte) MAC address
+	deviceIDBytes, base64Err := base64.StdEncoding.DecodeString(deviceID)
```

Comment:
Actually on the unit testing note @oxtoacart I think `logging.go` and `errors.go` are the #1 and #2 files in terms of lines *not covered* by tests. Looking at https://coveralls.io/builds/7207505/source?filename=src%2Fgithub.com%2Fgetlantern%2Fflashlight%2Flogging%2Flogging.go, it seems like things like `EnableFileLogging` and `initContext` would be pretty quick wins in terms of just adding potentially pretty trivial tests.

I know the tls proxy stuff and also just getting this in today (actually I think getting a version out today is probably even more important than tlsproxy ATM) are super high priority, but just noticed that while you're in the area we might be able to take another bite out of the untested lines in coverage.